MapboxCoreNavigation/NavigationRouteOptions.swift

```diff
@@ -0,0 +1,52 @@
+import Foundation
+import MapboxDirections
+
+/**
+ A `NavigationRouteOptions` object is a structure that specifies the criteria for results returned by the Mapbox Directions API.
+ 
+ `NavigationRouteOptions` is a subclass of `RouteOptions` that has been optimized for navigation. Pass an instance of this class into the `Directions.calculate(_:completionHandler:)` method.
```

Comment:
This is kind of silly, but what if we extended Directions to make NavigationRouteOptions a little more discoverable to folks who don’t read documentation?

```swift
extension Directions {
    @objc(calculateDirectionsWithNavigationOptions:completionHandler:)
    open func calculate(_ options: NavigationOptions, completionHandler: @escaping CompletionHandler) -> URLSessionDataTask {
        return calculate(options as RouteOptions, completionHandler)
    }
}
```

Disclaimer: I haven’t tried this; just playing around in a text box in a browser.